upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -662,34 +662,55 @@ func deregisterInstanceFromClassicLoadBalancer(c AWSCloud, loadBalancerNames []s
 // deregisterInstanceFromTargetGroups ensures that instances are fully unused in the corresponding targetGroups before instance termination.
 // this ensures that connections are fully drained from the instance before terminating.
 func deregisterInstanceFromTargetGroups(c AWSCloud, targetGroupArns []string, instanceId string) error {
-	klog.Infof("Deregistering instance from targetGroups: %v", targetGroupArns)
+	eg, _ := errgroup.WithContext(context.Background())
+
+	for _, targetGroupArn := range targetGroupArns {
+		arn := targetGroupArn
```

Comment:
required to avoid `targetGroupArn` changing underlying value and ultimately passing the wrong value to `deregisterInstanceFromTargetGroup`.
See https://eli.thegreenplace.net/2019/go-internals-capturing-loop-variables-in-closures/ for more info.